packages/medusa/src/services/sales-channel.ts

```diff
@@ -35,8 +38,36 @@ class SalesChannelService extends TransactionBaseService<SalesChannelService> {
     this.eventBusService_ = eventBusService
   }
 
-  async retrieve(id: string): Promise<SalesChannel> {
-    throw new Error("Method not implemented.")
+  async retrieve(
+    salesChannelId: string,
+    config: FindConfig<SalesChannel> = {}
+  ): Promise<SalesChannel> {
+    return await this.atomicPhase_(async (manager) => {
+      const salesChannelRepo = manager.getCustomRepository(
+        this.salesChannelRepository_
+      )
+
+      const { relations, ...query } = buildQuery(
+        {
+          id: salesChannelId,
+        },
+        config
+      )
+
+      const saleChannel = await salesChannelRepo.findOneWithRelations(
```

Comment:
todo, very nit: Naming
```suggestion
      const salesChannel = await salesChannelRepo.findOneWithRelations(
```